pkg/cli/config/refreshcabundle/refresh_ca_bundle.go

```diff
@@ -0,0 +1,181 @@
+package refreshcabundle
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type RefreshCABundleOptions struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+
+	ConfigAccess clientcmd.ConfigAccess
+	ClusterName  string
+
+	// for convenience. You can compare to an original by doing.
+	// oc config view --raw -o jsonpath='{.clusters[?(@.name == "the-name")].cluster.certificate-authority-data}' | base64 -d
+	DryRun bool
+
+	genericclioptions.IOStreams
+}
+
+var (
+	setClusterLong = templates.LongDesc(i18n.T(`
+		Update the CA bundle by reading the content from an OpenShift cluster.`))
+
+	setClusterExample = templates.Examples(`
+		# Refresh the CA bundle for the current context's cluster
+		oc config refresh-ca-bundle
+
+		# Refresh the CA bundle for the cluster named e2e in your kubeconfig
+		oc config refresh-ca-bundle e2e
+
+		# Print the CA bundle from the current OpenShift cluster's apiserver.
+		oc config refresh-ca-bundle --dry-run`)
+)
+
+func NewRefreshCABundleOptions(restClientGetter genericclioptions.RESTClientGetter, configAccess clientcmd.ConfigAccess, streams genericclioptions.IOStreams) *RefreshCABundleOptions {
+	return &RefreshCABundleOptions{
+		RESTClientGetter: restClientGetter,
+		ConfigAccess:     configAccess,
+		IOStreams:        streams,
+	}
+}
+
+func NewCmdConfigRefreshCABundle(restClientGetter genericclioptions.RESTClientGetter, configAccess clientcmd.ConfigAccess, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewRefreshCABundleOptions(restClientGetter, configAccess, streams)
+
+	cmd := &cobra.Command{
+		Use:                   fmt.Sprintf("refresh-ca-bundle [NAME]"),
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Update the OpenShift CA bundle by contacting the apiserver."),
+		Long:                  setClusterLong,
+		Example:               setClusterExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd))
+			cmdutil.CheckErr(o.Run(context.TODO()))
+
+		},
+	}
+
+	cmd.Flags().BoolVar(&o.DryRun, "dry-run", o.DryRun, "display the CA bundle, but don't make any changes to the kubeconfig")
+
+	return cmd
+}
+
+func (o RefreshCABundleOptions) Run(ctx context.Context) error {
+	err := o.Validate()
+	if err != nil {
+		return err
+	}
+
+	clientConfig, err := o.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	kubeClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
```

Comment:
the client setup should probably be a part of `Complete()` as fake client is easier to mock than the whole `genericclioptions.RESTClientGetter` for unit tests